src/java.base/unix/native/libjli/java_md_solinux.c

```diff
@@ -816,6 +817,11 @@ JVMInit(InvocationFunctions* ifn, jlong threadStackSize,
         int argc, char **argv,
         int mode, char *what, int ret)
 {
+#if defined(__FreeBSD__) && defined(PROC_STACKGAP_CTL)
+    /* Must disable the kernel stack guard pages before threads are created */
+    int arg = PROC_STACKGAP_DISABLE | PROC_STACKGAP_ENABLE_EXEC;
```

Comment:
If PROC_STACKGAP_ENABLE_EXEC enables PROT_EXEC on the stack, I'm pretty sure this is not needed. At least with the gcc toolchain `"-Wl,-z,noexecstack` is used. See `make/autoconf/flags-ldflags.m4`

Reply:
Good catch.  I was thinking that when kib first posted it but then forgot when I was cut & pasting it.  I've removed the use of PROC_STACKGAP_ENABLE_EXEC and pushed that change.